pkg/server/catalog/catalog.go

```diff
@@ -122,11 +122,6 @@ func Load(ctx context.Context, config Config) (_ *Repository, err error) {
 	repo := &Repository{
 		log: config.Log,
 	}
-	defer func() {
```

Comment:
I think the correct behavior here is to keep the defer and instead change the Close() method to test if the dataStoreCloser and catalogClosers are set before trying to close them individually. This defer cleans up and closes the datastore connection on configuration error.

Reply:
That was my first fix, but then I thought that was redundant. Thanks for pointing the reason, now I see the point.